### ISSUE TITLE
Setting `spec.json` `global_settings.skull_strip_algorithm` to `none` does not work

### ISSUE THREAD

**Issue by cchaswell** [HALFpipe/HALFpipe#69]
How can I pass options for fmriprep through halfpipe? I received some data that is already brain-extracted and didn't realize so the outputs are wacky. I wanted to add the command "--skull-strip-t1w skip" for fmriprep but am not sure how I can do that. I tried just adding it to the singularity command but had an error. Is there a way to do this?

singularity run \
--cleanenv \
--bind /:/ext \
/mnt/BIAC/duhsnas-pri.dhe.duke.edu/morey_biac/Lab/pipeline2/halfpipe_latest.sif \
--workdir /ext/mnt/BIACnfs/duhsnas-pri.dhe.duke.edu/morey_biac/Lab/Masaryk1 \
--only-run \
--execgraph-file /ext/mnt/BIACnfs/duhsnas-pri.dhe.duke.edu/morey_biac/Lab/Masaryk1/execgraph.30_chunks.f7947966.pickle.xz \
--only-chunk-index ${SGE_TASK_ID} \
--nipype-n-procs 2 \
--skull-strip-t1w skip \
--nipype-omp-nthreads 8 \
--verbose

**Comment by HippocampusGirl** [comment]
In this case you need to edit the `spec.json` file and change `"skull_strip_algorithm": "ants"` to `"skull_strip_algorithm": "none"` if all are skull stripped, or `"skull_strip_algorithm": "auto"` if you want fMRIPrep to try to detect for each image if skull stripping is needed.

**Comment by HippocampusGirl** [comment]
HALFpipe code completely skips the fMRIPrep command line, so the command line flags have no effect

**Comment by cchaswell** [comment]
Ok thanks! Is this only for T1s or also the functionals? The T1s have been stripped but the resting state scans have not been.

**Comment by HippocampusGirl** [comment]
Functional scans will always be skull stripped, I don't think this can be disabled :-)

**Comment by cchaswell** [comment]
I changed skull_strip_algorithm to none, but it still seems like the program is skull stripping (from the log and what the report looks like). The brains are poorly skull-stripped to begin with and we are working to get original T1s. Do I need to change any other setting (like remove skull-strip template?). 

I can't attach the whole file due to size but inside it says:

[2021-03-25 13:31:36,0282] [nipype.workflow ] [INFO     ] [MultiProc] Running 0 tasks, and 119 jobs ready. Free memory (GB): 113.10/113.10, Free processors: 2/2.
[2021-03-25 13:31:36,0634] [nipype.workflow ] [INFO     ] [Node] Setting-up "nipype.fmriprep_wf.single_subject_C0453_wf.func_preproc_task_rest_wf.final_boldref_wf.enhance_and_skullstrip_bold_wf.init_aff" in "/ext/mnt/BIACnfs/duhsnas-pri.dhe.duke.edu/morey_biac/Lab/Masaryk1/nipype/fmriprep_wf/single_subject_C0453_wf/func_preproc_task_rest_wf/final_boldref_wf/enhance_and_skullstrip_bold_wf/init_aff".
[2021-03-25 13:31:36,0731] [nipype.workflow ] [INFO     ] [MultiProc] Running 1 tasks, and 118 jobs ready. Free memory (GB): 112.90/113.10, Free processors: 0/2.
<img width="1506" alt="Screen Shot 2021-03-29 at 9 42 38 AM" src="https://user-images.githubusercontent.com/40609098/112848123-ba884c00-9075-11eb-8b20-97024d485dad.png">

**Comment by HippocampusGirl** [comment]
That seems like a bug, I will look into it

**Comment by cchaswell** [comment]
Lea, is there any update on this? It seems that I can only get T1s from this site that are already skull stripped.

**Comment by l-espana** [comment]
I don't know if this is helpful for this issue or not, but I have an interesting use case... We have multiple sites and ran the same spec.json pipeline on all, but there were a few here and there that had bad skullstrips in two separate sites. We did a quick skullstrip ourselves and then reran the same pipeline without skullstripping. 
For one run setting the skull_strip_algorithm to "none" _did_ work, but for a separate run with different subjects, it didn't work. I noticed that the fmriprep.config.toml in the nipype directory did not have the skull_strip_t1w set to "skip". I just reran after resetting some of these things and re-generating the graphs through halfpipe, and now they have run correctly. 
Perhaps somewhere along the line it doesn't always get set in the fmriprep toml file? Just some extra info for ya'll.
